pypesto/sampling/theano.py

```diff
@@ -10,31 +12,90 @@
     tt = NullType = None
 
 
+class CachedObjective:
+    """
+    Wrapper around an Objective which computes the gradient at each evaluation,
+    caching it for later calls.
+    Caching is only enabled after the first time the gradient is asked for
+    and disabled whenever the cached gradient is not used,
+    in order not to increase computation time for derivative-free samplers.
+
+    Parameters
+    ----------
+    objective:
+        The `pypesto.Objective` to wrap.
+    """
+
+    def __init__(self, objective: Objective):
```

Comment:
@yannikschaelte Only took 6 days for someone to use `Objective` when he intended to use `ObjectiveBase` ...

Reply:
Ah, looks like it already happened in https://github.com/ICB-DCM/pyPESTO/pull/351/files, maybe we should rethink whether it doensn't make more sense to call the base class `Objective`.

Reply:
I will change it